src/PostgREST/DbRequestBuilder.hs

```diff
@@ -343,7 +345,7 @@ returningCols rr@(Node _ forest) = returnings
     -- However if the "client_id" is present, e.g. mutateRequest to /projects?select=client_id,name,clients(name)
     -- we would get `RETURNING client_id, name, client_id` and then we would produce the "column reference \"client_id\" is ambiguous"
     -- error from PostgreSQL. So we deduplicate with Set:
-    returnings = S.toList . S.fromList $ fldNames ++ (colName <$> fkCols)
+    returnings = S.toList . S.fromList $ fldNames ++ (colName <$> fkCols) ++ pkCols
```

Comment:
I think the pkCols change is good. But we could get lost later, the returningCols function is getting a bit complicated. A comment explaining the pkCols inclusion would be nice.